benches/speed_tests.rs

```diff
@@ -83,177 +82,430 @@ pub fn poly_mul(c: &mut Criterion) {
+    for input_len in [10, 100, 1_000] {
+        group.bench_with_input(
+            BenchmarkId::new("serial", input_len),
+            &input_len,
+            |b, input_len| {
+                let vdaf = Prio3::new_sum_vec(num_shares, 1, *input_len).unwrap();
+                let measurement = (0..u128::try_from(*input_len).unwrap())
+                    .map(|i| i & 1)
+                    .collect::<Vec<_>>();
+                let nonce = black_box([0u8; 16]);
+                let verify_key = black_box([0u8; 16]);
+                let (public_share, input_shares) = vdaf.shard(&measurement, &nonce).unwrap();
+                b.iter(|| {
+                    vdaf.prepare_init(&verify_key, 0, &(), &nonce, &public_share, &input_shares[0])
+                        .unwrap()
+                });
+            },
+        );
+    }
 
     #[cfg(feature = "multithreaded")]
     {
-        let prio3 = Prio3::new_sum_vec_multithreaded(num_shares, 1, len).unwrap();
-        let measurement = vec![0; len];
-        group.bench_function(BenchmarkId::new("countvec_parallel", len), |b| {
-            b.iter(|| {
-                prio3.shard(&measurement, &nonce).unwrap();
-            })
-        });
+        for input_len in [10, 100, 1_000] {
+            group.bench_with_input(
+                BenchmarkId::new("parallel", input_len),
+                &input_len,
+                |b, input_len| {
+                    let vdaf = Prio3::new_sum_vec_multithreaded(num_shares, 1, *input_len).unwrap();
+                    let measurement = (0..u128::try_from(*input_len).unwrap())
+                        .map(|i| i & 1)
+                        .collect::<Vec<_>>();
+                    let nonce = black_box([0u8; 16]);
+                    let verify_key = black_box([0u8; 16]);
+                    let (public_share, input_shares) = vdaf.shard(&measurement, &nonce).unwrap();
+                    b.iter(|| {
+                        vdaf.prepare_init(
+                            &verify_key,
+                            0,
+                            &(),
+                            &nonce,
+                            &public_share,
+                            &input_shares[0],
+                        )
+                        .unwrap()
+                    });
+                },
+            );
+        }
     }
+    group.finish();
 
-    #[cfg(feature = "experimental")]
-    {
-        let len = 1000;
-        let prio3 = Prio3::new_fixedpoint_boundedl2_vec_sum(num_shares, len).unwrap();
-        let fp_num = fixed!(0.0001: I1F15);
-        let measurement = vec![fp_num; len];
-        group.bench_function(BenchmarkId::new("fixedpoint16_boundedl2_vec", len), |b| {
-            b.iter(|| {
-                prio3.shard(&measurement, &nonce).unwrap();
-            })
-        });
+    let mut group = c.benchmark_group("prio3histogram_shard");
+    for input_len in [10, 100, 1_000] {
```

Comment:
IIRC our practical usage of Prio3Histogram includes larger histogram sizes, approaching 10^4 and 10^5. Could consider benchmarking larger lengths?